pkg/cmd/roachtest/tests/copyfrom.go

```diff
@@ -0,0 +1,162 @@
+	l_receiptdate   DATE NOT NULL,
+	l_shipinstruct  CHAR(25) NOT NULL,
+	l_shipmode      CHAR(10) NOT NULL,
+	l_comment       VARCHAR(44) NOT NULL,
+	l_dummy         CHAR(1),
+	PRIMARY KEY     (l_orderkey, l_linenumber));
+CREATE INDEX  l_ok ON lineitem    (l_orderkey);
+CREATE INDEX  l_pk ON lineitem    (l_partkey);
+CREATE INDEX  l_sk ON lineitem    (l_suppkey);
+CREATE INDEX  l_sd ON lineitem    (l_shipdate);
+CREATE INDEX  l_cd ON lineitem    (l_commitdate);
+CREATE INDEX  l_rd ON lineitem    (l_receiptdate);
+CREATE INDEX  l_pk_sk ON lineitem (l_partkey, l_suppkey);
+CREATE INDEX  l_sk_pk ON lineitem (l_suppkey, l_partkey);
+`
+
+func initTest(ctx context.Context, t test.Test, c cluster.Cluster, sf int) {
+	if runtime.GOOS == "linux" {
+		if err := repeatRunE(
+			ctx, t, c, c.All(), "update apt-get", `sudo apt-get -qq update`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		if err := repeatRunE(
+			ctx,
+			t,
+			c,
+			c.All(),
+			"install dependencies",
+			`sudo apt-get install -qq postgresql`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		csv := fmt.Sprintf(tpchLineitemFmt, sf)
+		c.Run(ctx, c.Node(1), "rm -f /tmp/lineitem-table.csv")
+		c.Run(ctx, c.Node(1), fmt.Sprintf("curl '%s' -o /tmp/lineitem-table.csv", csv))
+	}
+}
+
+func runTest(ctx context.Context, t test.Test, c cluster.Cluster, pg string) {
+	start := time.Now()
+	det, err := c.RunWithDetailsSingleNode(ctx, t.L(), c.Node(1), fmt.Sprintf(`cat /tmp/lineitem-table.csv | %s -c "COPY lineitem FROM STDIN WITH CSV DELIMITER '|';"`, pg))
+	if err != nil {
+		t.L().Printf("%v\n", det.Stdout)
+		t.L().Printf("%v\n", det.Stderr)
+		t.Fatal(err)
+	}
+	dur := time.Since(start)
+	t.L().Printf("%v\n", det.Stdout)
+	rows := 0
+	copy := ""
+	_, err = fmt.Sscan(det.Stdout, &copy, &rows)
+	require.NoError(t, err)
+	rate := int(float64(rows) / dur.Seconds())
+	t.L().Printf("results: %d rows/s", rate)
+	// Write the copy rate into the stats.json file to be used by roachperf.
+	c.Run(ctx, c.Node(1), "mkdir", t.PerfArtifactsDir())
+	cmd := fmt.Sprintf(
+		`echo '{ "copy_rate": %d }' > %s/stats.json`,
+		rate, t.PerfArtifactsDir(),
+	)
+	c.Run(ctx, c.Node(1), cmd)
+}
+
+func runCopyFromPG(ctx context.Context, t test.Test, c cluster.Cluster, sf int) {
+	initTest(ctx, t, c, sf)
+	c.Run(ctx, c.Node(1), "sudo -i -u postgres psql -c 'DROP TABLE IF EXISTS lineitem'")
+	c.Run(ctx, c.Node(1), fmt.Sprintf("sudo -i -u postgres psql -c '%s'", lineitemSchema))
+	runTest(ctx, t, c, "sudo -i -u postgres psql")
+}
+
+func runCopyFromCRDB(ctx context.Context, t test.Test, c cluster.Cluster, sf int) {
+	c.Put(ctx, t.Cockroach(), "./cockroach", c.All())
+	c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings(), c.All())
+	initTest(ctx, t, c, sf)
+	urls, err := c.InternalPGUrl(ctx, t.L(), c.Node(1))
+	require.NoError(t, err)
+	m := c.NewMonitor(ctx, c.All())
+	m.Go(func(ctx context.Context) error {
+		// psql w/ url first are doesn't support --db arg so have to do this.
```

Comment:
sorry could you elaborate on what this means?

Reply:
Basically you can use the "conninfo" ie url first arg or use the -U/-d/-h arguments but not both so I had to stick the db name into the url otherwise it tried to connect to db "$USER".